providers/fips/fipsprov.c

```diff
@@ -1266,85 +1266,23 @@ static void deferred_deinit(void)
     }
 }
 
-struct deferred_test_status {
-    const char *alg;
-    int pass;
-};
-
-static struct deferred_test_status deferred_kat_integrity_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_cipher_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_asym_cipher_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_asym_keygen_tests[] = {
-    { "SLH-DSA-SHA2-128f", 0 },
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_kem_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_digest_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_signature_tests[] = {
-    { "SLH-DSA-SHA2-128f", 0 },
-    { "SLH-DSA-SHAKE-128f", 0 },
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_kdf_tests[] = {
-    { NULL, 0 }
-};
-static struct deferred_test_status deferred_kat_ka_tests[] = {
-    { NULL, 0 }
-};
-
-static struct deferred_test_status *deferred_tests_status[FIPS_DEFERRED_MAX] = {
-    deferred_kat_integrity_tests,
-    deferred_kat_cipher_tests,
-    deferred_kat_asym_cipher_tests,
-    deferred_kat_asym_keygen_tests,
-    deferred_kat_kem_tests,
-    deferred_kat_digest_tests,
-    deferred_kat_signature_tests,
-    deferred_kat_kdf_tests,
-    deferred_kat_ka_tests,
-};
-
-static int FIPS_kat_deferred(OSSL_LIB_CTX *libctx, int type, const char *alg)
+static int FIPS_kat_deferred(OSSL_LIB_CTX *libctx, FIPS_DEFERRED_TEST *test)
 {
-    int idx = 0;
-    int ret = 0;
+    int ret = FIPS_DEFERRED_TEST_FAILED;
 
     if (!CRYPTO_THREAD_run_once(&deferred_once, deferred_init))
-        return 0;
+        return FIPS_DEFERRED_TEST_FAILED;
 
     if (deferred_lock == NULL)
-        return 0;
-
-    if (type > FIPS_DEFERRED_MAX)
-        return 0;
-
-    for (; deferred_tests_status[type][idx].alg != NULL; idx++) {
-        if (strcmp(deferred_tests_status[type][idx].alg, alg) == 0)
-            break;
-    }
-    if (deferred_tests_status[type][idx].alg == NULL)
-        return 0;
-
-    if (deferred_tests_status[type][idx].pass)
-        return 1;
+        return FIPS_DEFERRED_TEST_FAILED;
 
     /*
-     * now that we found the test, before we do anything else make sure
-     * a local test is not already in progress or we'll deadlock
+     * before we do anything, make sure * a local test is not already in
```

Comment:
NOTE: I already fixed this stray asterisk locally and will push it after the next round of reviews with any requested changes